src/domain/fiat/index.ts

```diff
@@ -15,3 +15,8 @@ export const OrderType = {
   Locked: "immediate",
   Active: "quote",
 } as const
+
+export const CENTS_PER_USD = 100
+
+export const add = <T extends number>(arg0: T, arg1: T): T => (arg0 + arg1) as T
+export const sub = <T extends number>(arg0: T, arg1: T): T => (arg0 - arg1) as T
```

Comment:
Careful - our `checkedToSats` method ensures that `sats > 0`. Casting here breaks that validation. Need to consider where all we are depending on sats being `>0` and wether or not this can be an issue or else we need to re-check in this method. Actually perhaps we should recheck regardless and throw an error (instead of returning) so that we will get visibility incase this breaks down at runtime.

Reply:
I tried to make it so that we could do `A.add(B).sub(C)` but wasn't able to code the type correctly. maybe you'll know how to do it?

Reply:
I'll do the check you suggest for now.

Reply:
> I tried to make it so that we could do A.add(B).sub(C) but wasn't able to code the type correctly. maybe you'll know how to do it?

This is an additional indication for a more 'complex` amount value that would host that function:

```
type MoneyAmount<T extends UsdCents | Satoshis> = {
  currency: T extends UsdCents ? "USD" : "BTC"
  baseAmount: T
  add: (other: MoneyAmount<T>) -> MoneyAmount<T>
  sub: (other: MoneyAmount<T>) -> MoneyAmount<T>
}
```
etc.

Reply:
ie: circular dependancies are not allowed